### PR TITLE
Update release-notes.md

### PR DESCRIPTION
**Pull request by heyitswei**
Updates for Proton and Cloud

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -2,6 +2,23 @@
 
 This page summarizes changes for each major update in Proton and Timeplus Cloud, including new features and important bug fixes. For previous release notes, please check [this page](changelog).
 
+## Apr 1, 2024
+
+*Timeplus Proton:*
+  * Added support for selecting message key from Apache Kafka external streams, via `SELECT _message_key, * FROM external_stream`.
+  * Streaming processing now supports nullable data type.
+  * [External Table](proton-clickhouse-external-table#create-external-table): When naming ClickHouse external tables, special characters such as dashes are now supported. Simply set `table='test-a-b'` in the `CREATE EXTERNAL STREAM` DDL.
```
Comment (jovezhong) [inline docs/release-notes.md:10]: DDL is wrong, should be `CREATE EXTERNAL TABLE`. "When naming ClickHouse tables" is confusing.  The user has already given a name in ClickHouse table, but if the name contains things like dash, previous version of Proton doesn't support it

File: docs/release-notes.md [hunk 2]
```diff
@@ -2,6 +2,23 @@
...
+  * [External Stream](proton-kafka#create-external-stream): Error handling and connection pooling/retry for Kafka external streams have been greatly improved.
+  * Materialized View: Added option to [skip dirty/unexpected data](query-syntax#settings). If you set `SETTINGS recovery_policy=best_effort`, Timeplus will try up to 3 times, then skip dirty data and continue processing the rest of the data.
+
+*Timeplus Cloud:*
```
Comment (jovezhong) [inline docs/release-notes.md:14]: From now on, let's call this Timeplus Cloud and Timeplus Enterprise

**Review by jovezhong — CHANGES_REQUESTED** [review]

File: docs/release-notes.md (hunk 2 above)
Comment (heyitswei) [reply docs/release-notes.md]: Thanks for your comments, @jovezhong. Both items now updated.

File: docs/release-notes.md [hunk 3]
```diff
@@ -2,6 +2,23 @@
...
 This page summarizes changes for each major update in Proton and Timeplus Cloud, including new features and important bug fixes. For previous release notes, please check [this page](changelog).
 
+## Apr 1, 2024
+
+*Timeplus Proton:*
+  * Added support for selecting message key from Apache Kafka external streams, via `SELECT _message_key, * FROM external_stream`.
+  * Streaming processing now supports nullable data type.
+  * [External Table](proton-clickhouse-external-table#create-external-table): ClickHouse external tables with names containing special characters (such as dashes) are now supported. Simply set `table='test-a-b'` in the `CREATE EXTERNAL TABLE` DDL.
+  * [External Stream](proton-kafka#create-external-stream): Error handling and connection pooling/retry for Kafka external streams have been greatly improved.
+  * Materialized View: Added option to [skip dirty/unexpected data](query-syntax#settings). If you set `SETTINGS recovery_policy=best_effort`, Timeplus will try up to 3 times, then skip dirty data and continue processing the rest of the data.
```
Comment (jovezhong) [inline docs/release-notes.md:12]: Need change again:
SETTINGS recovery_policy='best_effort'

**Review by jovezhong — APPROVED** [review]

**Pull request merged**